OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs

```diff
@@ -405,6 +405,20 @@ public void OnInfiltrated(Actor a, LuaFunction func)
 			GetScriptTriggers(a).RegisterCallback(Trigger.OnInfiltrated, func, Context);
 		}
 
+		[Desc("Call a function when this actor is discovered by an enemy (doesn't work for neutral actors). " +
+			"The callback function will be called as func(Actor self, Player discoverer)")]
+		public void OnDiscovered(Actor a, LuaFunction func)
+		{
+			GetScriptTriggers(a).RegisterCallback(Trigger.OnDiscovered, func, Context);
+		}
+
+		[Desc("Call a function when this player is discovered by an enemy (doesn't work for neutrals). " +
+			"The callback function will be called as func(Actor self, Player discoverer)")]
```

Comment:
But he doesn't? The actor that was discovered is returned, not the PlayerActor. I specifically changed that in https://github.com/penev92/OpenRA/pull/5.

And you can get the owner by checking `actor.Owner` in your callback function, so this way is more flexible.